text/0000-road-to-typescript.md

```diff
@@ -0,0 +1,160 @@
+---
+Stage: Accepted
+Start Date: 2021-02-19
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): Ember.js, Ember Data, Ember CLI, Learning, Steering
+RFC PR: 
+---
+
+<!--- 
+Directions for above: 
+
+Stage: Leave as is
+Start Date: Fill in with today's date, YYYY-MM-DD
+Release Date: Leave as is
+Release Versions: Leave as is
+Relevant Team(s): Fill this in with the [team(s)](README.md#relevant-teams) to which this RFC applies
+RFC PR: Fill this in with the URL for the Proposal RFC PR
+-->
+
+# Road to TypeScript
+
+## Summary
+
+TypeScript has become a key part of the front-end development ecosystem over the past several years, and powers many of the best developer experiences in the front-end ecosystem. Ember was a relatively early TypeScript adopter for its internals, and there is widespread usage in the ecosystem with community support, but to date Ember has not provided “out of the box” or official support for TypeScript. This RFC declares our intent to make TypeScript a first-class citizen of the Ember ecosystem, as a peer to JavaScript which makes the developer experience better for *all* Ember developers. It outlines the key constraints and goals for the effort, and outlines a roadmap for accomplishing those goals.
```

Comment:
>  but to date Ember has not provided “out of the box” or official support for TypeScript

might be good to clarify that this is about "for app / addon developers"? not sure that is or isn't a good distinction but wanted to point it out in case you do want to tweak

Reply:
Resolved in 28a1fc4 – I like clarifying this!